test/test_jit.py

```diff
@@ -2383,6 +2393,63 @@ def forward(self, x, seq_lens):
         f = io.BytesIO()
         torch.onnx._export(m, (x, seq_lens), f, verbose=False)
 
+    def test_script_outputs(self):
+        with self.assertRaisesRegex(RuntimeError, "value cannot be used as a tuple"):
+            @torch.jit.script
+            def foo(a):
+                c, d = a + a
+                return c + d
+
+        @torch.jit.script
+        def return3():
+            return 1, 2, 3
+
+        with self.assertRaisesRegex(RuntimeError, "too many values to unpack"):
+            @torch.jit.script
+            def bind2():
+                a, b = return3()
+                print(a)
+                print(b)
+
+    def test_script_chunk(self):
+        @torch.jit.script
+        def foo(a):
+            b, c = torch.chunk(a, dim=0, chunks=2)
+            return b
+        v = torch.rand(10, 3)
+        self.assertEqual(torch.chunk(v, dim=0, chunks=2)[0], foo(v))
+
+        with self.assertRaisesRegex(RuntimeError, "too many values to unpack"):
+            @torch.jit.script
+            def foo(a):
+                b, c = torch.chunk(a, dim=0, chunks=3)
+                return b
+
+    def test_tuples(self):
+        @torch.jit.script
+        def foo(i):
+            a = torch.chunk(i, dim=0, chunks=2)
+            c = a
+            # some nonsense with if-statements and loops to check
+            # that tuple lowering doesn't fail
+            if True:
+                c = torch.chunk(i, dim=0, chunks=2)
+            t0, t1 = c
+            while False:
+                t0, t1 = c
+                c = torch.chunk(i, dim=0, chunks=2)
+            return t0
+
+        v = torch.rand(10, 3)
+        self.assertEqual(torch.chunk(v, dim=0, chunks=2)[0], foo(v))
+
+        with self.assertRaisesRegex(RuntimeError, "variable 'a' previously has type"):
```

Comment:
We're diverging from Python semantics here, I think. Is this alright?

Reply:
It's ok. We're only a subset, and it's impossible to figure out a type of this variable if we don't know a priori whether the `if` branch will run.

Reply:
Yeah, thankfully this fully fits into "a subset of programs, which don't change the type of their variables, will work"